src/study/system-model/library.cpp

```diff
@@ -89,6 +89,15 @@ LibraryBuilder& LibraryBuilder::withModels(std::vector<Model>&& models)
     return *this;
 }
 
+LibraryBuilder& LibraryBuilder::withModelsMap(std::unordered_map<std::string, Model>&& models)
+{
+    for (auto&& pair: models)
+    {
+        library_.models_->insert(std::make_pair(pair.first, std::move(pair.second)));
```

Comment:
```suggestion
        library_.models_->insert(std::move(pair));
```
seems to work as well considering function signatures (`unordered_map&&`, `auto&& pair`)